cpp/ql/src/semmle/code/cpp/dataflow/internal/FlowVar.qll

```diff
@@ -90,8 +90,13 @@ cached class FlowVar extends TFlowVar {
 private module PartialDefinitions {
   private newtype TPartialDefinition =
     TExplicitFieldStoreQualifier(Expr qualifier, ControlFlowNode node) {
-      exists(FieldAccess fa |
-        isInstanceFieldWrite(fa, node) and qualifier = fa.getQualifier()
+      exists(FieldAccess fa | qualifier = fa.getQualifier() |
+        isInstanceFieldWrite(fa, node)
+        or
+        exists(PartialDefinition pd |
+          node = pd.getSubBasicBlockStart() and
+          fa = pd.getDefinedExpr()
+        )
```

Comment:
I'm struggling with this as the comments explaining `PartialDefinitions` are rather minimal.  Perhaps we could talk through an example?

Reply:
Thanks, I understand what's going on now.